### PR TITLE
Require PHP >= 8.1

### PR DESCRIPTION
**Pull request by ben-challis**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -44,7 +44,7 @@
     "bin-dir": "bin",
     "sort-packages": true,
     "platform": {
-      "php": "8.0.13"
+      "php": "8.1.15"
```
Comment (marmichalski) [inline composer.json:47]: Please update https://github.com/Lendable/clock/blob/94dc38fc942ee4578f6195a15c21afcf4122560d/Dockerfile.php-cli#L1

File: lib/Date.php [hunk 2]
```diff
@@ -8,11 +8,11 @@
...
 final class Date
 {
-    private int $year;
+    private readonly int $year;
 
-    private int $month;
+    private readonly int $month;
 
-    private int $day;
+    private readonly int $day;
```
Comment (marmichalski) [inline lib/Date.php:15]: Is there any reason this was not moved to ctor property promotion? :see_no_evil:

Comment (ben-challis) [reply lib/Date.php:15]: No, it's just what Rector did, I'll update.

Comment (ben-challis) [reply lib/Date.php]: https://github.com/Lendable/clock/pull/427/commits/f500c5274499bec49736c3ee1aaa539f70ec47b0

**Review by marmichalski — APPROVED** [review]
:medal_sports:

**Pull request merged**